File: src/main/kotlin/codeacademy/bookingforum/app/comment/CommentMapper.java
package codeacademy.bookingforum.app.comment;

import codeacademy.bookingforum.app.post.Post;
import codeacademy.bookingforum.app.user.auth.UserAuth;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;

@Service
public class CommentMapper {
    public CommentDto toDto(Comment entity) {
        if (entity == null) {
            return null;
        }
        CommentDto dto = new CommentDto();

        dto.setId(entity.getId());
        dto.setContent(entity.getContent());
        dto.setDateCommented(entity.getDateCommented());
        dto.setLikes(entity.getLikes());
        dto.setDislikes(entity.getDislikes());
        dto.setUserId(entity.getUser().getId());
        dto.setPostId(entity.getPost().getId());

        return dto;
    }

    public Comment fromDto(CommentDto dto) {
        if (dto == null) {
            return null;
        }
        Comment entity = new Comment();

        entity.setId(dto.getId());
        entity.setContent(dto.getContent());
        if (dto.getDateCommented() == null) {
            entity.setDateCommented(LocalDateTime.now());
        } else {
            entity.setDateCommented(dto.getDateCommented());
        }
        entity.setLikes(dto.getLikes());
        entity.setDislikes(dto.getDislikes());
        entity.setUser(new UserAuth(dto.getUserId()));
        entity.setPost(new Post(dto.getPostId()));

        return entity;
    }
}

File: src/main/kotlin/codeacademy/bookingforum/app/comment/CommentMapper.kt
package codeacademy.bookingforum.app.comment

import codeacademy.bookingforum.app.post.Post
import codeacademy.bookingforum.app.user.auth.UserAuth
import org.springframework.stereotype.Service

@Service
class CommentMapper {

    fun fromDtoToEntity(dto: CommentDto?): Comment {
        return Comment(
            id = dto?.id,
            dateCommented = dto?.dateCommented,
            content = dto?.content,
            postId = Post(dto?.postId),
            userId = UserAuth(dto?.userId),
            likes = dto?.likes,
            dislikes = dto?.dislikes
        )
    }

    fun fromEntityToDto(entity: Comment?): CommentDto {
        return CommentDto(
            id = entity?.id,
            dateCommented = entity?.dateCommented,
            content = entity?.content,
            postId = entity?.postId?.id,
            userId = entity?.userId?.id,
            likes = entity?.likes,
            dislikes = entity?.dislikes,
        )
    }

    fun fromEntityListToDtoList(entityList : List<Comment>?): List<CommentDto>? {
        return if(entityList.isNullOrEmpty()){
            val listOfCommentDto : MutableList<CommentDto> = emptyList<CommentDto>().toMutableList()
            if(entityList?.isEmpty() == false){
                entityList.forEach {
                    listOfCommentDto.add(fromEntityToDto(it))
                  }
            }
            listOfCommentDto
        }else{
            null
        }
    }
}